### PR TITLE
Restyle Job Runs

### PR DESCRIPTION
**Pull request by NavyAdmiral**
For [#163407115](https://www.pivotaltracker.com/story/show/163407115)

Used Typography from Activity for consistency.  
Overall, I think it looks similar to what the story has asked.  

------------------------------------------
PS: we need a monospace font (have to manually adjust width without it)

### CODE REVIEW

**Comment by dimroc** [comment]
We do lose insight on this change, notable the type of `pending` status.
We have

1. pending_confirmations
2. pending_sleep
3. pending_connection
4. pending_bridge

Can we work the pending type in?

**Comment by NavyAdmiral** [comment]
@dimroc How about this?
image

**Comment by NavyAdmiral** [comment]
@dimroc pushing commits for this pic that I liked the most
image

**Comment by dimroc** [comment]
I'm surprised we don't account for the difference between pending and in_progress. Let's stray from the design slightly and display "In Progress" for that state.

This way, you can just normalize all states by titlizing them from snake case, with the exception of `error` and `complete`:

```
const statusText = status => {
  if (status === 'error') return 'Failed'
  if (status === 'completed') return 'Complete'
  return titleize(status)
}
```

"in_progress" => "In Progress"

We can take inspiration from https://github.com/sindresorhus/titleize#readme although it doesn't account for the underscore.

**Comment by NavyAdmiral** [comment]
@dimroc Does it look good now?
image

**Comment by NavyAdmiral** [comment]
Github not letting us disable `force-pushed from ... to ...` is the worst thing 😢

**Comment by thoughtmerchant** [comment]
Looks good but please justify the run status to the right of the table cell rather than centering. It makes it easier to scan.

**Comment by NavyAdmiral** [comment]
image
@thoughtmerchant Your wish is my command 🔮

**Comment by NavyAdmiral** [comment]
Other than the magic numbers here https://github.com/smartcontractkit/chainlink/pull/1052/files#diff-d7d7fa8b5328d1017697ef719bc214caR35 and not using Grid, is there anything terribly wrong with the PR?

**Comment by thoughtmerchant** [comment]
@NavyAdmiral love it! Thank you.

**Comment by NavyAdmiral** [comment]
@rupurt How do I set up fetch-mock? I can't debug the failing test because I'm getting `rror: Uncaught [Error: fetch-mock: No fallback response defined for GET to 6688/v2/specs/c60b9927eeae43168ddbe92584937b1b]`

**Comment by rupurt** [comment]
This is an example usage https://github.com/smartcontractkit/chainlink/blob/master/gui/__tests__/containers/Jobs/Show.test.js#L34

This is where it's setup https://github.com/smartcontractkit/chainlink/blob/master/jest.setup.js#L13

**Comment by NavyAdmiral** [comment]
@rupurt Yea I'm aware of both, but for some reason both on Linux and Windows my jest is always failing on global.fetch usages:
```
 ● containers/Jobs/Index › can page through the list of jobs

    fetch-mock: No fallback response defined for GET to 6688/v2/specs?page=1&size=1
```

Since it's failing for me on both environments and fresh on just `yarn jest` I assume I'm doing something wrong

**Comment by dimroc** [comment]
Fixed this up for you @NavyAdmiral. Two issues:
1. In the test, runs didn't have a status, so the `components/JobRuns/List.js` was blowing up on `r.status.startsWith(...)` saying undefined has no method `startsWith`.
2. Running `yarn test-ci` hid the errors. Not very friendly and it might have to do with the broken invocation being in the view. This command will run the test and give you a real call stack for the error:

```
cross-env NODE_ENV=test yarn jest gui/__tests__/containers/Jobs/Show.test.js
```

File: gui/src/components/JobRuns/List.js [hunk 1]
```diff
@@ -56,18 +56,31 @@ const styles = theme => {
...
     .replace(/(?:^|\s|-)\S/g, x => x.toUpperCase())
+}
 
 const statusText = status => {
   if (status === 'errored') return 'Failed'
   if (status === 'completed') return 'Completed'
   return titleize(status)
 }
 
+const classFromStatus = (classes, status) => {
```
Comment (NavyAdmiral) [inline gui/src/components/JobRuns/List.js]: 🥂

**Comment by NavyAdmiral** [comment]
@dimroc Makes sense, thanks for the commits. However, even with these commits and passing CI, running `yarn test-ci` or `cross-env NODE_ENV=test yarn jest gui/__tests__/containers/Jobs/Show.test.js` still errors for me with:
```js
  containers/Jobs/Show
    × renders the details of the job spec and its latest runs (102ms)
    × displays a view more link if there are more runs than the display count (25ms)

  ● containers/Jobs/Show › renders the details of the job spec and its latest runs

    fetch-mock: No fallback response defined for GET to 6688/v2/specs/c60b9927eeae43168ddbe92584937b1b

      36 | const get = (path, query) =>
      37 |   global
    > 38 |     .fetch(formatURI(path, query), { credentials: 'include' })
         |      ^
      39 |     .then(parseResponse)
      40 | 
      41 | const post = (path, body) => {
```

**Comment by NavyAdmiral** [comment]
In `global.fetch.getOnce('/v2/user/balances', accountBalanceResponse)` how does fetch-mock know the full url is `http://localhost:6688/v2/user/balances`? It seems to me that my test aren't able to tell what is the url

**Comment by rupurt** [comment]
@NavyAdmiral I've bumped the `fetch-mock` version [here](https://github.com/smartcontractkit/chainlink/pull/1065). Can you give that a try and report your results?

**Comment by rupurt** [comment]
There are several matching strategies that `fetch-mock` uses. This is what it looks like on the latest version https://github.com/wheresrhys/fetch-mock/blob/master/src/lib/generate-matcher.js

**Comment by NavyAdmiral** [comment]
@rupurt That doesn't fix it, unfortunately. We can merge the PR since that is a problem on my end and unrelated to this PR.

File: gui/src/components/JobRuns/List.js [hunk 2]
```diff
@@ -2,58 +2,122 @@ import React from 'react'
...
+      paddingLeft: theme.spacing.unit * 1.5,
+      paddingRight: theme.spacing.unit * 1.5,
+      paddingTop: theme.spacing.unit / 2,
+      paddingBottom: theme.spacing.unit / 2,
+      borderRadius: theme.spacing.unit * 2,
+      width: 'fit-content',
+      display: 'inline-block'
+    },
+    failed: {
+      backgroundColor: '#e9faf2',
```
Comment (rupurt) [inline gui/src/components/JobRuns/List.js]: These colors should be pulled from the theme

File: gui/src/components/JobRuns/List.js [hunk 3]
```diff
@@ -2,58 +2,122 @@ import React from 'react'
...
+    completed: {
+      backgroundColor: '#e9faf2',
+      color: '#4ed495'
+    }
   }
-})
+}
+
+// insight from - https://github.com/sindresorhus/titleize
+const titleize = input => {
```
Comment (rupurt) [inline gui/src/components/JobRuns/List.js]: I feel like this should live in `gui/src/utils`. Can you also add a unit test for this?

**Comment by dimroc** [comment]
Hey @NavyAdmiral, not sure what command you're using to run tests, but simply type `jest gui/` and you'll get all the errors that are somehow missed with `yarn test-ci`.

**Pull request merged**